src/com/seafile/seadroid2/ui/activity/UnlockGesturePasswordActivity.java

```diff
@@ -106,6 +114,11 @@ public void onPatternDetected(List<LockPatternView.Cell> pattern) {
                 mLockPatternView
                         .setDisplayMode(LockPatternView.DisplayMode.Correct);
                 settingsMgr.setupGestureLock();
+                if (extras != null) {
+                    Intent sIntent = new Intent(UnlockGesturePasswordActivity.this, ShareToSeafileActivity.class);
+                    sIntent.putExtras(extras);
+                    UnlockGesturePasswordActivity.this.startActivity(sIntent);
+                }
```

Comment:
The `if` block should be removed now. We don't need to treat ShareToSeafileActivity differently.